fix(gke-auth-plugin): Prevent hang and add --gke-auth-plugin option

plinkr opened the pull request:
When connecting to GKE clusters using `gke-gcloud-auth-plugin` authentication, `kl` would hang indefinitely if the `gke-gcloud-auth-plugin` binary was not present in the system's PATH. This occurred without any error message, forcing users to manually terminate the process.

This commit addresses the issue by:
- Introducing a new error message that is displayed when the `gke-gcloud-auth-plugin` is required but cannot be found in the system's PATH.
- Adding a new CLI option, `--gke-auth-plugin`, which allows users to explicitly specify the directory where the `gke-gcloud-auth-plugin` binary is located. This is particularly useful when the plugin is not in the default system PATH.
- When a custom path is provided, it is prepended to the `PATH` environment variable for the current process, ensuring that `kl` and `client-go` can locate the plugin. This addresses the issue https://github.com/robinovitch61/kl/issues/9

plinkr committed:
fix(gke-auth-plugin): Prevent hang and add --gke-auth-plugin option

When connecting to GKE clusters using `gke-gcloud-auth-plugin` authentication, `kl` would hang indefinitely if the `gke-gcloud-auth-plugin` binary was not present in the system's PATH. This occurred without any error message, forcing users to manually terminate the process.

This commit addresses the issue by:
- Introducing a new error message that is displayed when the `gke-gcloud-auth-plugin` is required but cannot be found in the system's PATH.
- Adding a new CLI option, `--gke-auth-plugin`, which allows users to explicitly specify the directory where the `gke-gcloud-auth-plugin` binary is located. This is particularly useful when the plugin is not in the default system PATH.
- When a custom path is provided, it is prepended to the `PATH` environment variable for the current process, ensuring that `kl` and `client-go` can locate the plugin.
This addresses the issue https://github.com/robinovitch61/kl/issues/9

robinovitch61 commented on internal/k8s/client/gke_auth_plugin.go:
```diff
@@ -0,0 +1,67 @@
+	errorMsg := fmt.Sprintf(
+		"gke-gcloud-auth-plugin not found in system PATH for context %s.\n"+
+			"You can either:\n"+
+			"  - %s and ensure 'google-cloud-sdk/bin' is in your system's PATH.\n"+
+			"  - Or provide the plugin directory via the --gke-auth-plugin flag to override the default lookup.\n",
```
Hey @plinkr , let's remove the second option - any new flags/args that we add we have to support forever to keep backwards compatibility, so let's not add a dependency on gke long term.

Instead, let's keep everything you have here, but just recommend that `google-cloud-sdk/bin` is in system's PATH if it's not. That way, users can add it to their PATH (permanently or temporarily) and then rerun `kl`, rather than running `kl` with a special flag to specify it's location.

plinkr replied:
Hi @robinovitch61, sounds good to me. I’ve already removed it in commit 620001bfe3facab137d05ddb57b0b84e6e776b02

robinovitch61 replied:
Awesome, sorry for the delay - will rereview shortly

robinovitch61 commented on cmd/cmd.go:
```diff
@@ -136,6 +136,10 @@ var (
 			cfgFileEnvVar: "since",
 			description:   `Show logs since startup time minus this duration. E.g. 5s, 2m, 1.5h, 2h45m. Default 1m`,
 		},
+		"gke-auth-plugin": {
+			cfgFileEnvVar: "gke-auth-plugin",
+			description:   `Path to the directory containing gke-gcloud-auth-plugin. Use this if the plugin is not in your system PATH when authenticating to GKE.`,
+		},
```
As mentioned in https://github.com/robinovitch61/kl/pull/10/files?diff=unified&w=0#r2172923705, let's remove this option

plinkr committed:
refactor: Remove `--gke-auth-plugin` flag and custom path lookup

The `--gke-auth-plugin` command-line argument and its associated logic have been removed.

Previously, this flag allowed users to specify a custom directory where the `gke-gcloud-auth-plugin` could be found. With this change, `kl` will now exclusively rely on the system's PATH to locate the `gke-gcloud-auth-plugin`
